manifests/params.pp

```diff
@@ -186,13 +186,16 @@
   $tftp_syslinux_files = undef
 
   # DHCP settings - requires optional DHCP puppet module
-  $dhcp                   = false
-  $dhcp_listen_on         = 'https'
-  $dhcp_managed           = true
-  $dhcp_interface         = 'eth0'
-  $dhcp_gateway           = '192.168.100.1'
-  $dhcp_range             = false
-  $dhcp_option_domain     = [$::domain]
+  $dhcp                    = false
+  $dhcp_split_config_files = true # smart-proxy 1.11+
+  $dhcp_listen_on          = 'https'
+  $dhcp_managed            = true
+  $dhcp_provider           = 'isc'
```

Comment:
The `dhcp_` prefix is currently prepended in the template, matching how `dns_provider` works.

I could change the default value to include `dhcp_` and then remove the prefix if it's an old-style config file (since the two vendor/provider parameters work equally for both old and new configs).